Code/Tools/AssetProcessor/native/ui/SourceAssetTreeModel.cpp

```diff
@@ -38,12 +38,51 @@ namespace AssetProcessor
         m_sourceToTreeItem.clear();
         m_sourceIdToTreeItem.clear();
 
-        m_sharedDbConnection->QuerySourceAndScanfolder(
-            [&](AzToolsFramework::AssetDatabase::SourceAndScanFolderDatabaseEntry& sourceAndScanFolder)
+        if (!m_intermediateAssets)
+        {
+            // AddOrUpdateEntry will remove intermediate assets if they shouldn't be included in this tree.
+            m_sharedDbConnection->QuerySourceAndScanfolder(
+                [&](AzToolsFramework::AssetDatabase::SourceAndScanFolderDatabaseEntry& sourceAndScanFolder)
+                {
+                    AddOrUpdateEntry(sourceAndScanFolder, sourceAndScanFolder, true);
+                    return true; // return true to continue iterating over additional results, we are populating a container
+                });
+        }
+        else
+        {
+            AzToolsFramework::AssetDatabase::ScanFolderDatabaseEntry scanFolderEntry;
+
+            if(!m_intermediateAssetFolderId.has_value())
```

Comment:
Why doesn't this use GetIntermediateAssetScanFolderId?

Reply:
It attempts to, in MainWindow.cpp line 428 and 429, but it wasn't actually available when this was being constructed.